src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
>You're fixated on your current implementation. 

And you ignore/forget that the `size()` here gives not the preferred size specified in .ui files.
Here it returns {100, 30} which is too small.

>Why do you need setting a default value every time?

Not I want that, only because the true intended value is populated then.

>Why do you need storing the default value in the config file?

what default value? the `size()` below returns the dialog size *after* the user resizes!